pkg/manifest/ostree_deployment.go

```diff
@@ -93,42 +101,67 @@ func (p *OSTreeDeployment) getBuildPackages(Distro) []string {
 
 func (p *OSTreeDeployment) getOSTreeCommits() []ostree.CommitSpec {
 	if p.ostreeSpec == nil {
-		panic("getOSTreeCommits(): ostree commit not set for pipeline")
+		return []ostree.CommitSpec{}
 	}
 	return []ostree.CommitSpec{*p.ostreeSpec}
 }
 
 func (p *OSTreeDeployment) getOSTreeCommitSources() []ostree.SourceSpec {
+	if p.commitSource == nil {
+		return []ostree.SourceSpec{}
+	}
 	return []ostree.SourceSpec{
-		p.commitSource,
+		*p.commitSource,
+	}
+}
+
+func (p *OSTreeDeployment) getContainerSpecs() []container.Spec {
+	if p.containerSpec == nil {
+		return []container.Spec{}
+	}
+	return []container.Spec{*p.containerSpec}
+}
+
+func (p *OSTreeDeployment) getContainerSources() []container.SourceSpec {
+	if p.containerSource == nil {
+		return []container.SourceSpec{}
+	}
+	return []container.SourceSpec{
+		*p.containerSource,
 	}
 }
 
 func (p *OSTreeDeployment) serializeStart(packages []rpmmd.PackageSpec, containers []container.Spec, commits []ostree.CommitSpec) {
-	if p.ostreeSpec != nil {
+	if p.ostreeSpec != nil && p.containerSpec != nil {
```

Comment:
Should this be `||`, it's invaldi to call `serializeStart` if either is set right?